scss/util/_components.scss

```diff
@@ -0,0 +1,164 @@
+@function foundation_component_enabled($key: '') {
+	@if (not map-has-key($foundation-components, $key)) {
+		@return false;
+	}
+
+	@if ($key == "grid" and map-get($foundation-components, "flex-grid")) {
+		@return false;
+	}
+	
+	@return map-get($foundation-components, $key);
+}
+
+@mixin foundation-components-include() {
+  @include foundation-global-styles;
+
+  @if (foundation_component_enabled("grid")) {
```

Comment:
In France, we call this an "If forest".
Because we can't `@include` dynamically, the only way I see to avoid it is to declare the components by using a `@mixin` checking if the component is enabled/disabled, and directly generating the CSS.
Well, almost like here but split into the component files.

Also, it is not possible anymore to easily customize the component generation order, or add its own components/code between. I think it is an important possibility that we shouldn't drop.